ffi/examples/read-table/arrow.c

```diff
@@ -163,24 +163,62 @@ static void visit_read_data(void* vcontext, ExclusiveEngineData* data)
   free(arrow_data); // just frees the struct, the data and schema are freed/owned by add_batch_to_context
 }
 
+static bool starts_with_n(const char *pre, size_t lenpre, const char *str, size_t lenstr)
+{
+    if (lenstr < lenpre) return false;
+    return memcmp(pre, str, lenpre) == 0;
+}
+
+typedef struct {
```

Comment:
Do we need this struct? `KernelStringSlice` is exactly this? Also, we save an unnecessary unpack.